BaseLib/TMPUtil.h

```diff
@@ -0,0 +1,46 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef BASELIB_TMPUTIL_H
+#define BASELIB_TMPUTIL_H
+
+namespace BaseLib
+{
+//! Has sequence of integers as template parameters
+template <int...>
+struct IntegerSequence {
+};
+
+//! Generates an IntegerSequence.
+//!
+//! \see http://stackoverflow.com/a/7858971
+template <int N, int... S>
+struct GenerateIntegerSequence {
+    // effectively pushes N-1 from the left to the list int... S of integers.
+    typedef typename GenerateIntegerSequence<N - 1, N - 1, S...>::type type;
+};
```

Comment:
Why not `std::integer_sequence` from C++14?  This is one more good reason to switch to C++14.